src/receive_imf.rs

```diff
@@ -2343,13 +2354,15 @@ async fn apply_group_changes(
         {
             let grpname = &sanitize_single_line(grpname);
             let old_name = &sanitize_single_line(old_name);
-            if chat_id
-                .update_timestamp(
-                    context,
-                    Param::GroupNameTimestamp,
-                    mime_parser.timestamp_sent,
-                )
-                .await?
+
+            let chat_group_name_timestamp = chat.param.get_i64(Param::GroupNameTimestamp);
+            let group_name_timestamp = group_name_timestamp.unwrap_or(mime_parser.timestamp_sent);
+            // NB: For old chats (created before "Chat-Group-Name-Timestamp" header introduction)
+            // this may cause an extra name update and `ChatModified` event, but that's fine.
+            if chat_group_name_timestamp < Some(group_name_timestamp)
```

Comment:
If timestamps are equal, we keep the local name. To provide consistency, we need to choose some name cmp function. E.g. in case of membership consistency, we prefer adding members rather than removing. Here we can just compare strings or hashes